links/src/main/java/juja/microservices/links/repository/impl/LinksRepositoryImpl.java

```diff
@@ -2,21 +2,41 @@
 
 import juja.microservices.links.model.Link;
 import juja.microservices.links.repository.LinksRepository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.stereotype.Repository;
 
+import javax.inject.Inject;
+import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Set;
 
 /**
  * @author Ivan Shapovalov
+ * @author Vladimid Zadorozhniy
  */
+
+@Repository
 public class LinksRepositoryImpl implements LinksRepository {
 
+    private String mongoCollectionName;
+
+    @Inject
+    private MongoTemplate mongoTemplate;
+
+    public LinksRepositoryImpl(MongoTemplate mongoTemplate, @Value("${spring.data.mongodb.collection}") String mongoCollectionName) {
+        this.mongoTemplate = mongoTemplate;
+        this.mongoCollectionName = mongoCollectionName;
+    }
+
     @Override
     public Link saveLink() {
         return null;
     }
 
     @Override
     public Set<Link> getAllLinks() {
-        return null;
+        List<Link> links = mongoTemplate.findAll(Link.class, mongoCollectionName);
```

Comment:
The same story I had implementing appropriate util class after an interface containing Set as a return type and finally I've changed it for the List after careful consideration. The only reason to use Set instead of List is an intention to avoid duplicate checking with a 'silent' link save while our case means explicit warning to the user which tries to save an already stored link (such a user won't be awarded with any 'thanks' provided by the gamification and they definitely have to be warned about it). On the other hand if we use List we get an ordered iterable collection widely used in similar situations (including previously accepted in-project util class realization). So if you don't mind to use List exactly could you please fix the implemented interface return type as well as other classes returning links collection?